packages/firebase_core/android/src/main/java/io/flutter/plugins/firebase/core/FirebaseCorePlugin.java

```diff
@@ -1,100 +1,31 @@
 // Copyright 2017 The Chromium Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
-
 package io.flutter.plugins.firebase.core;
 
-import android.content.Context;
-import com.google.firebase.FirebaseApp;
-import com.google.firebase.FirebaseOptions;
-import io.flutter.plugin.common.MethodCall;
+import androidx.annotation.NonNull;
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
 import io.flutter.plugin.common.MethodChannel;
-import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
-import io.flutter.plugin.common.MethodChannel.Result;
 import io.flutter.plugin.common.PluginRegistry;
-import java.lang.String;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
 
-public class FirebaseCorePlugin implements MethodCallHandler {
+public class FirebaseCorePlugin implements FlutterPlugin {
+  private static final String CHANNEL_NAME = "plugins.flutter.io/firebase_core";
```

Comment:
nits(non-blocking): this can be refactored into `FirebaseCoreHandler` as well.

Reply:
I combined the `FlutterPlugin` and `MethodCallHandler` for PR simplicity.